### PR TITLE
GCP plugin: integration tests

### PR DESCRIPTION
**Pull request by tam7t**
**What this PR does / why we need it**:

Adds integration tests for the GCP Secret Manager plugin: https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Starting with a single test case.

Changes to test-infra: https://github.com/kubernetes/test-infra/pull/19512

and uploading of a new secret (`gcp-secrets-store-cred`) to prow

### CODE REVIEW

File: test/bats/gcp.bats [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_TESTS_DIR=test/bats/tests/gcp
+WAIT_TIME=60
+SLEEP_TIME=1
+NAMESPACE=default
+PROVIDER_YAML=https://raw.githubusercontent.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/tam7t/v010/deploy/provider-gcp-plugin.yaml
```
Comment (aramase) [inline test/bats/gcp.bats:9]: Should this be the url be https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/blob/main/deploy/provider-gcp-plugin.yaml?

File: Makefile [hunk 2]
```diff
@@ -150,7 +150,7 @@ else
 			--set linux.image.repository="e2e/secrets-store-csi" \
 			--set linux.image.tag=$(IMAGE_VERSION) \
 			--set linux.image.pullPolicy="IfNotPresent" \
-			--set grpcSupportedProviders=azure \
+			--set grpcSupportedProviders=azure,gcp \
```
Comment (aramase) [inline Makefile:153]:
```suggestion
			--set grpcSupportedProviders="azure;gcp" \
```

File: test/bats/gcp.bats (hunk 1 above)
Comment (tam7t) [reply test/bats/gcp.bats:9]: Yeah, this branch has the correct build referenced in the yaml where the `main` branch does not have it yet. I will update this PR once I merge https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/pull/59 and push a new release

**Comment by aramase** [comment]
@tam7t Could you squash the commits?

**Review by aramase — COMMENTED** [review]
/lgtm

File: test/bats/gcp.bats (hunk 1 above)
Comment (tam7t) [reply test/bats/gcp.bats]: Updated for the `main` branch.

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**